integration/helpers/archive.go

```diff
@@ -0,0 +1,137 @@
+/*
+ * Teleport
+ * Copyright (C) 2024  Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package helpers
+
+import (
+	"archive/tar"
+	"archive/zip"
+	"compress/gzip"
+	"io"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"runtime"
+
+	"github.com/gravitational/teleport"
+	"github.com/gravitational/trace"
+)
+
+// CompressDirToZipFile compresses directory into a `.zip` format.
+func CompressDirToZipFile(sourcePath, destPath string) error {
+	archive, err := os.Create(destPath)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	defer archive.Close()
+
+	zipWriter := zip.NewWriter(archive)
+	err = filepath.Walk(sourcePath, func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			return trace.Wrap(err)
+		}
+		if info.IsDir() {
+			return nil
+		}
+		file, err := os.Open(path)
+		if err != nil {
+			return trace.Wrap(err)
+		}
+		zipFileWriter, err := zipWriter.Create(filepath.Base(path))
```

Comment:
Is it intentional to flatten all of the files into the top-level directory of the zip file?

Reply:
yes, since this one for tests and I changed logic for unarchive so now it is not important where tsh, tctl located inside of archive and identified by application name suffix

Reply:
Could you clarify that in the godoc? It would help a ton if each exported function described its behavior and what you need to pass for each argument.